crates/ruff/src/rules/pydocstyle/rules/sections.rs

```diff
@@ -1044,10 +1050,31 @@ fn google_section(checker: &mut Checker, docstring: &Docstring, context: &Sectio
             checker.diagnostics.push(diagnostic);
         }
     }
+}
 
-    if checker.settings.rules.enabled(Rule::UndocumentedParam) {
-        if matches!(context.kind, SectionKind::Args | SectionKind::Arguments) {
-            args_section(checker, docstring, context);
+fn parse_google_sections(checker: &mut Checker, lines: &[&str], docstring: &Docstring) {
+    let mut documented_args: FxHashSet<String> = FxHashSet::default();
+    let section_contexts = &section_contexts(lines, SectionStyle::Google);
+    for section_context in section_contexts {
+        // Checks occur at the section level. Since two sections (args/keyword args and their
+        // variants) can list arguments, we need to unify the sets of arguments mentioned in both
+        // then check for missing arguments at the end of the section check.
+        if matches!(
+            section_context.kind,
+            SectionKind::Args
+                | SectionKind::Arguments
+                | SectionKind::KeywordArgs
+                | SectionKind::KeywordArguments
+                | SectionKind::OtherArgs
+                | SectionKind::OtherArguments
+        ) {
+            documented_args.extend(args_section(section_context));
         }
+
+        google_section(checker, docstring, section_context);
+    }
+
+    if checker.settings.rules.enabled(Rule::UndocumentedParam) {
+        missing_args(checker, docstring, &documented_args);
```

Comment:
Previously, this only ran if the `Args` section was present (so if `Args` was omitted entirely, we didn't raise an error). Is that still the case?

Reply:
Just wondering - is there a reason we prioritize `Args` so highly?

Reply:
Ah sorry, I wasn't clear -- what I meant was: we should only trigger if at least _one_ arguments-related section is present. It doesn't have to be `Args`.